master/internal/rm/kubernetesrm/spec.go

```diff
@@ -129,6 +132,103 @@ func (p *pod) configureEnvVars(
 	return envVars, nil
 }
 
+func (p *pod) configureProxyResources() []gatewayProxyResource {
+	if p.exposeProxyConfig == nil {
+		return nil
+	}
+
+	var resources []gatewayProxyResource
+	// TODO(RM-275/gateways) think about experiments, should they proxy every pod, or only rank 0?
+	for i, proxyPort := range p.req.ProxyPorts {
+		// TODO(RM-276/gateways) use something more reasonable for the name.
+		// Podname is too long currently. There is like a 63 characeter limit for DNS.
+		// We should really be under this for everything even if not required (like it is for
+		// service and TCPRoute). Bradley is changing this but hopefully we land before him.
+		tooLong := fmt.Sprintf("porti%d-%s-%s",
+			i, p.submissionInfo.taskSpec.Description, uuid.New().String())
+		sharedName := tooLong[:min(63, len(tooLong)-1)]
+
+		allocLabels := map[string]string{
+			determinedLabel: p.submissionInfo.taskSpec.AllocationID,
+		}
+
+		serviceSpec := &k8sV1.Service{
+			ObjectMeta: metaV1.ObjectMeta{
+				Name:      sharedName,
+				Namespace: p.namespace,
+				Labels:    allocLabels,
+			},
+			Spec: k8sV1.ServiceSpec{
+				Ports: []k8sV1.ServicePort{
+					{
+						Protocol: k8sV1.ProtocolTCP,
+						Port:     int32(proxyPort.Port),
+					},
+				},
+				Selector: allocLabels,
```

Comment:
👍🏼  yeah this is a better mapping to a single allocation than what I had. would we gain anything by going a level higher to selecting by taskid? what multi-port task do we wanna support

Reply:
I don't think so? I think it probably wouldn't matter since the main case, ntsC allocationID is effectively the same as taskID.

Maybe in the generic experiment ticket we will learn more for it to make sense one way or the other